backend/de.metas.business/src/main/java/de/metas/shipping/PurchaseOrderToShipperTransportationService.java

```diff
@@ -78,9 +81,17 @@ public static PurchaseOrderToShipperTransportationService newInstanceForUnitTest
 
 	private static final String AD_PROCESS_VALUE_C_Order_SSCC_Print_Jasper = "C_Order_SSCC_Print_Jasper";
 
+	public void addMaterialReceiptCandidatesToShipperTransportation(@NonNull final ShipperTransportationId shipperTransportationId, @NonNull final Collection<OrderAndLineId> orderAndLineIds)
+	{
+		final ImmutableListMultimap<I_C_Order, I_C_OrderLine> orderToLinesMap = orderDAO.getOrderToLinesMap(orderAndLineIds);
+
+		orderToLinesMap.keySet()
+				.forEach(order -> addPurchaseOrderLines(shipperTransportationId, order, orderToLinesMap.get(order)));
+	}
+
 	public void addPurchaseOrdersToShipperTransportation(@NonNull final ShipperTransportationId shipperTransportationId, @NonNull final IQueryFilter<I_C_Order> queryFilter)
 	{
-		final ImmutableList<OrderId> validPurchaseOrdersIds = Services.get(IQueryBL.class)
+		final ImmutableList<OrderId> validPurchaseOrdersIds = queryBL
```

Comment:
hurts to see this queryBL call here and not in OrderDAO :(

I am fine to leave it as it is because it was not introduced in this issue, but
* pls be very aware that this is a BAD example
* don't use it as a precedent